tests/Microsoft.Data.DataFrame.Tests/DataFrameTests.cs

```diff
@@ -0,0 +1,507 @@
+                verify = newColumn.Equals(df1.Column(i) * 2);
+                Assert.Equal(true, verify[0]);
+
+                newColumn = df1[df1.Column(i).Name] - df2[df2.Column(i).Name];
+                verify = newColumn.Equals(0);
+                Assert.Equal(true, verify[0]);
+
+                newColumn = df1[df1.Column(i).Name] * df2[df2.Column(i).Name];
+                verify = newColumn.Equals(df1.Column(i) * df1.Column(i));
+                Assert.Equal(true, verify[0]);
+
+                var df1Column = df1.Column(i) + 1;
+                var df2Column = df2.Column(i) + 1;
+                newColumn = df1Column / df2Column;
+                verify = newColumn.Equals(1);
+                Assert.Equal(true, verify[0]);
+
+                newColumn = df1Column % df2Column;
+                verify = newColumn.Equals(0);
+                Assert.Equal(true, verify[0]);
+
+                newColumn = df1[df1.Column(i).Name] == df2[df2.Column(i).Name];
+                verify = newColumn.Equals(true);
+                Assert.Equal(true, verify[0]);
+
+                newColumn = df1[df1.Column(i).Name] != df2[df2.Column(i).Name];
+                verify = newColumn.Equals(false);
+                Assert.Equal(true, verify[0]);
+
+                newColumn = df1[df1.Column(i).Name] >= df2[df2.Column(i).Name];
+                verify = newColumn.Equals(true);
+                Assert.Equal(true, verify[0]);
+
+                newColumn = df1[df1.Column(i).Name] <= df2[df2.Column(i).Name];
+                verify = newColumn.Equals(true);
+                Assert.Equal(true, verify[0]);
+
+                newColumn = df1[df1.Column(i).Name] > df2[df2.Column(i).Name];
+                verify = newColumn.Equals(false);
+                Assert.Equal(true, verify[0]);
+
+                newColumn = df1[df1.Column(i).Name] < df2[df2.Column(i).Name];
+                verify = newColumn.Equals(false);
+                Assert.Equal(true, verify[0]);
+            }
+        }
+
+        [Fact]
+        public void TestBinaryOperationsWithConversions()
+        {
+            DataFrame df = DataFrameTests.MakeDataFrameWithTwoColumns(10);
+
+            // Add a double to an int column
+            DataFrame dfd = df.Add(5.0f);
+            var dtype = dfd.Column(0).DataType;
+            Assert.True(dtype == typeof(double));
+
+            // Add a decimal to an int column
+            DataFrame dfm = df.Add(5.0m);
+            dtype = dfm.Column(0).DataType;
+            Assert.True(dtype == typeof(decimal));
+
+            // int + bool should throw
+            Assert.Throws<System.NotSupportedException>(() => df.Add(true));
+
+            var dataFrameColumn1 = new PrimitiveColumn<double>("Double1", Enumerable.Range(0, 10).Select(x => (double)x));
+            df.SetColumn(0, dataFrameColumn1);
+            // Double + comparison ops should throw
+            Assert.Throws<System.NotSupportedException>(() => df.And(true));
```

Comment:
You should be able to `using System;` at the top and then you don't need the full `System.NotSupportedException` here.